### ISSUE TITLE
Extra stats for cached queries (fullCount)

### ISSUE THREAD

**Issue by matcho** [arangodb/arangodb#4546]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 3.3

Mode:
- [x] Single-Server

Storage-Engine:
- [x] rocksdb

On this operating system:
  - [x] Ubuntu .deb

### this is an AQL-related issue:

I'm issuing AQL via:
- [x] HTTP API
- [x] this Driver: arangojs

The issue can be reproduced using any dataset.

When query cache is enabled, no `extra` data is returned in the cursor. I understand this behaviour, as the query was not executed but just searched in the cache.

However, when asking for options like `fullCount`, it is quite problematic that the cached query result does not include the `fullCount` property… if a program relies on this value, does it mean that the only solution is to disable caching for this query ? What if the query takes a long time or is executed very often ?

Instead, in such a case could the `stats` be cached along with the query results ?

Many thanks,
Mathias

Steps to reproduce the issue :
##### arangosh
```
require("@arangodb/aql/cache").properties({ mode: "on" });
```
##### HTTP API - first run (the issue occurs with arangojs driver too)
```
curl -X POST --data-binary @- --dump - http://login:pass@server:8529/_db/mydb/_api/cursor <<EOF
{
  "query": "FOR doc IN mycollection LIMIT 10 RETURN doc",
  "options": { "fullCount": true }
}
EOF

{
  "result":[…],
  "hasMore":false,
  "cached":false,
  "extra": {
    "stats": {
      "writesExecuted":0,
      "writesIgnored":0,
      "scannedFull":1645720,
      "scannedIndex":0,
      "filtered":0,
      "httpRequests":0,
      "fullCount":1645720,
      "executionTime":1.5409233570098877
    },
    "warnings": []
  },
  "error":false,
  "code":201
}
```
##### HTTP API - second run
```
curl -X POST --data-binary @- --dump - http://login:pass@server:8529/_db/mydb/_api/cursor <<EOF
{
  "query": "FOR doc IN mycollection LIMIT 10 RETURN doc",
  "options": { "fullCount": true }
}
EOF

{
  "result":[…],
  "hasMore":false,
  "cached":true,
  "extra": {},
  },
  "error":false,
  "code":201
}
```

**Comment by matcho** [comment]
Hello,

I'm sorry to insist on this feature request, but it's quite important : at the moment any query relying on "fullCount" result has to be issued without cache, which has a terrible impact on performance…

Do you think this could be included in a future release ?

Thank you,
Mathias

**Comment by jsteemann** [comment]
A proof-of-concept PR for this is in the works now:
https://github.com/arangodb/arangodb/pull/6360

**Comment by jsteemann** [comment]
Implemented for 3.4.
